bind/flag.go

```diff
@@ -279,17 +279,25 @@ func HTTPServerConfig(fs *pflag.FlagSet, cfg *forwarder.HTTPServerConfig, prefix
 	fs.VarP(anyflag.NewValueWithRedact[*url.Userinfo](cfg.BasicAuth, &cfg.BasicAuth, forwarder.ParseUserinfo, RedactUserinfo),
 		namePrefix+"basic-auth", "", "<username[:password]>"+
 			"Basic authentication credentials to protect the server. ")
+}
 
-	httpLogModes := []httplog.Mode{
-		httplog.None,
-		httplog.ShortURL,
-		httplog.URL,
-		httplog.Headers,
-		httplog.Body,
-		httplog.Errors,
+func HTTPLogConfig(fs *pflag.FlagSet, logcfg *httplog.Config) {
+	valid := make([]string, 0, len(logcfg.Modules()))
+	for _, m := range logcfg.Modules() {
```

Comment:
There is no need to copy it here.
If you want a copy for safety, just copy the slice before returning it in Modules.

Reply:
It's not just a copy though, it's removing the empty string.

If we don't do that we get a usage string like `[<>|<api>:]` or `[<>]`. I believe the latter case confuses Cobra and it doesn't print proper usage text.

Or do you mean do the filtering inside the `Modules()` function?